src/garage/tf/regressors/gaussian_mlp_regressor_model.py

```diff
@@ -178,15 +179,67 @@ def _build(self, state_input, name=None):
 
         normalized_xs_var = (state_input - x_mean_var) / x_std_var
 
-        normalized_mean, normalized_log_std, std_param, dist = super()._build(
+        _, normalized_dist_mean, normalized_dist_log_std = super()._build(
             normalized_xs_var)
 
+        # Since regressor expects [N, *dims], we need to squeeze the extra
+        # dimension
+        normalized_dist_log_std = tf.squeeze(normalized_dist_log_std, 1)
+
         with tf.name_scope('mean_network'):
-            means_var = normalized_mean * y_std_var + y_mean_var
+            means_var = normalized_dist_mean * y_std_var + y_mean_var
 
         with tf.name_scope('std_network'):
-            log_stds_var = normalized_log_std + tf.math.log(y_std_var)
+            log_stds_var = normalized_dist_log_std + tf.math.log(y_std_var)
+
+        normalized_dist = tfp.distributions.MultivariateNormalDiag(
+            loc=normalized_dist_mean,
+            scale_diag=tf.exp(normalized_dist_log_std))
+
+        vanilla_dist = tfp.distributions.MultivariateNormalDiag(
+            loc=means_var, scale_diag=tf.exp(log_stds_var))
+
+        return (normalized_dist, normalized_dist_mean, normalized_dist_log_std,
+                vanilla_dist, means_var, log_stds_var, x_mean_var, x_std_var,
+                y_mean_var, y_std_var)
+
+    def clone(self, name):
+        """Return a clone of the model.
 
-        return (means_var, log_stds_var, std_param, normalized_mean,
-                normalized_log_std, x_mean_var, x_std_var, y_mean_var,
-                y_std_var, dist)
+        It only copies the configuration of the primitive,
```

Comment:
I think we also need to copy the parameters.

Reply:
Will do in another PR.